### PR TITLE
feat(config): outputPath function support

### PR DESCRIPTION
**Pull request by xDisfigure**
# Feature 

## Request
Config: outputPath function support

### Expected 
I would like to have the control over the outputPath value so it can directly writes changes to a specific locale file.

```typescript
type outputPath = string | (locale: string, namespace: string) => string
```

### ℹ️   Important to consider
 **That option will only be available for javascript babel config file**

### Context & Usage 
In our application locales cannot be extracted into the same folder because they are shared across the app.
Its only after our build we copy every locales to the same folder (`public/locales/`)

**Example: Folder structure**
```
.
└── src/
    ├── locales/
    │   ├── en/
    │   └── fr/
    │       └── common.json
    ├── pages/
    │   └── Dashboard/
    │       └── locales/
    │           ├── en/
    │           └── fr/
    │               └── dashboard.json
    └── anotherFolder/
        └── locales/
            ├── en/
            └── fr/
                └── anotherFolder.json
```

**Example: babel.config.js**
```js
module.exports = {
  plugins: [
    [
      'i18next-extract',
      {
        locales: ['en', 'fr'],
        outputPath: (locale, namespace) => {
          const relPath = `src/**/*/locales/${locale}/${namespace}.json`
          const absPath = `${__dirname}/${relPath}`
          const results = glob.sync(absPath)

          if (!results.length) {
            throw new Error(`I18next: missing file in ${relPath}`)
          }

          return results[0]
        }
      }
    ]
  ],
}
```

### CODE REVIEW

**Comment by damln** [comment]
Thank you 🙏  Useful feature

File: tests/runner.ts [hunk 1]
```diff
@@ -79,10 +85,14 @@ function* genTestData(): IterableIterator<TestData> {
         '.extracted',
         testFileEnt.name.replace(/\.json$/, ''),
       );
+
       const outputPath =
-        testData.pluginOptions && testData.pluginOptions.outputPath
-          ? path.join(extractionDir, testData.pluginOptions.outputPath)
-          : path.join(extractionDir, 'translations.{{ns}}.{{locale}}.json');
+        testData.pluginOptions &&
```
Comment (gilbsgilbs) [inline tests/runner.ts:90]: Could you refactor this to use conditionals instead of ternaries here? Quite hard to follow the nested ternaries.

**Review by gilbsgilbs — CHANGES_REQUESTED** [review]
Thanks. PR looks clean and simple, congratulations. Could you update the documentation? I think you only need to update [this file](https://github.com/gilbsgilbs/babel-plugin-i18next-extract/blob/d680b407ebe9879b5a31f44453d01437ca007353/docs/templates/configuration.md.yml#L181). Please specify that you need a JS babel config for this to work.

File: tests/runner.ts (hunk 1 above)
Comment (xDisfigure) [reply tests/runner.ts:90]: > Thanks. PR looks clean and simple, congratulations. Could you update the documentation? I think you only need to update [this file](https://github.com/gilbsgilbs/babel-plugin-i18next-extract/blob/d680b407ebe9879b5a31f44453d01437ca007353/docs/templates/configuration.md.yml#L181). Please specify that you need a JS babel config for this to work.

going to :)

Comment (xDisfigure) [reply tests/runner.ts]: @gilbsgilbs done :)

**Comment by gilbsgilbs** [comment]
Thanks for the PR :tada: . I'll try to make a release today.

**Comment by xDisfigure** [comment]
> Thanks for the PR 🎉 . I'll try to make a release today.

with pleasure, thanks for your fast reply !

**Comment by gilbsgilbs** [comment]
Seems like updating dependencies isn't as straightforward as I originally intended. No runtime issues, but looks like babel's type guards don't work as expected anymore. I need to figure out what's happening (at least to pin the relevant libraries) before shipping a new release. Can't do it just yet sorry, we'll have to wait a bit (unless somebody takes a look at it).

I've opened #169 for tracking purposes.

**Comment by gilbsgilbs** [comment]
Should be released on npm under 0.8.0 tag. Thanks again for the contribution.
